src/query/functions/src/scalars/arithmetic.rs

```diff
@@ -147,79 +151,39 @@ pub fn register(registry: &mut FunctionRegistry) {
 
                         {
                             type T = F64;
-                            registry.register_2_arg_core::<NumberType<L>, NumberType<R>, NumberType<T>, _, _>(
+                            registry.register_passthrough_nullable_2_arg::<NumberType<L>, NumberType<R>, NumberType<T>, _, _>(
                                 "divide",
                                 FunctionProperty::default(),
                                 |_, _| FunctionDomain::MayThrow,
                                 vectorize_with_builder_2_arg::<NumberType<L>, NumberType<R>,  NumberType<T>>(
-                                    |a, b, output, _| {
+                                    |a, b, output, ctx| {
                                         let b = (b.as_() : T);
                                         if std::intrinsics::unlikely(b == 0.0) {
-                                            return Err("/ by zero".to_string());
+                                            ctx.set_error(output.len(), "/ by zero");
```

Comment:
If an error occurs, does the function need to continue to execute?

Reply:
Yes, it just continues in a single batch which is not very costly.